Improved docs on background jobs on instances

991jo opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #20488

<!--
    Please include a summary of the proposed changes below.
-->

Commit:
improved docs on running jobs with an instance

arthanson reviewed (changes requested):
@991jo can you please provide an example for this, also lower in the file under Scheduled Jobs the example there does enqueue with instance=self and not derived from JobsMixin which either the example is wrong or the wording here is not clear - can you please take a look.

Also, here is a bit reworded description that might be a bit clearer?

A Job can be executed for a specific instance of a Model. To enable this functionality, the model must include the JobsMixin.

When enqueuing a Job, you can associate it with a particular instance by passing that instance to the instance parameter.

Commit:
improved Background Jobs documentation

- added improved wording from @arthanson
- added missing JobsMixin in the example code

991jo commented:
Yes, the JobsMixin is missing in the example. I added it there.

I think your description is good. I replaced mine with yours.

arthanson reviewed (approved):
Thanks @991jo
